server/rollback.go

```diff
@@ -25,25 +33,88 @@ application.
 		RunE: func(cmd *cobra.Command, args []string) error {
 			ctx := GetServerContextFromCmd(cmd)
 			cfg := ctx.Config
+
+			deletePendingBlock, err := cmd.Flags().GetBool(FlagDeletePendingBlock)
+			if err != nil {
+				return err
+			}
+
+			if deletePendingBlock {
+				if err := deletePendingBlockIfExists(cfg); err != nil {
```

Comment:
probably don't need this feature in 0.46, since all the nodes with be running with the tendermint version that "validate block before persisting".